### PR TITLE
Fix #6673 Altering ref entity results in broken table

### LINKED ISSUE #6673: Altering ref entity results in broken table
### How to Reproduce
Upload [vkgl-model.xlsx](https://github.com/molgenis/molgenis/files/1317795/vkgl-model.xlsx)
Open the 'Consensus' Entity in the metadata manager.
Change ref entity in 'OMIM disease' attribute to another entity

### Expected behavior
Another entity is referenced and when the column is empty or the references are correct this will go fine

### Observed behavior
Unknown error!
Spinner
Table not usable (is fixed by changing the refentity back)

### PR DESCRIPTION
**Pull request by Mark-de-Haan**
#### Checklist
- [x] Functionality works & meets specifications
- [x] Code reviewed
- [x] Code unit/integration/system tested
- [x] User documentation updated
- [x] (If you have changed REST API interface) view-swagger.ftl updated
- [x] Test plan template updated; No but i did it for you, otherwise the test plan would fail now
- [x] Clean commits

### CODE REVIEW

**Comment by ConnorStroomberg** [comment]
when running `yarn test` i get the following output:

```
/karma-phantomjs-launcher" plugin:
  Path must be a string. Received null
```

**Comment by Mark-de-Haan** [comment]
Should be a local problem, because yarn test works on my machine.
I use yarn v1.0.2

File: molgenis-metadata-manager/src/main/frontend/src/store/actions.js [hunk 1]
```diff
@@ -170,6 +170,10 @@ export default {
    * Persist metadata changes to the database
    */
   [SAVE_EDITOR_ENTITY_TYPE] ({commit, state}: { commit: Function, state: State }) {
+    if (!state.editorEntityType.labelAttribute) {
+      commit(UPDATE_EDITOR_ENTITY_TYPE, {key: 'labelAttribute', value: state.editorEntityType.idAttribute})
```
Comment (ConnorStroomberg) [inline molgenis-metadata-manager/src/main/frontend/src/store/actions.js]: Changing the form input on save seems confusing to me, If a label is needed for entity to be valid, i suggest making it a required field ( both in back- and frontend ) and notifying the user about the need for a label attribute.

File: molgenis-metadata-manager/src/main/frontend/src/components/MetadataManagerEntityEditForm.vue [hunk 2]
```diff
@@ -123,7 +123,13 @@
     name: 'metadata-manager-entity-edit-form',
     methods: {
       saveEntityType () {
-        this.$store.dispatch(SAVE_EDITOR_ENTITY_TYPE)
+        if (this.editorEntityType.idAttribute === null || this.editorEntityType.idAttribute === undefined) {
+          this.$store.commit(CREATE_ALERT, {type: 'warning', message: 'ID attribute can not be empty'})
+        } else if (this.editorEntityType.labelAttribute === null || this.editorEntityType.labelAttribute === undefined) {
+          this.$store.commit(CREATE_ALERT, {type: 'warning', message: 'Label attribute can not be empty'})
```
Comment (ConnorStroomberg) [inline molgenis-metadata-manager/src/main/frontend/src/components/MetadataManagerEntityEditForm.vue]: This is good, but does not stop the entity being created on the server side. The code creating the entity should check the modal validity. The checking model validity on the client side could maybe be grouped into a dedicated named function.

**Pull request merged**